scipy/optimize/tests/test__differential_evolution.py

```diff
@@ -849,6 +850,7 @@ def constr_f2(x):
             assert_almost_equal(cv, np.array([[0.0, 0.0, 0.], [2.1, 4.2, 0]]))
             assert cv.shape == (2, 3)
 
+    @pytest.mark.parallel_threads(1)
```

Comment:
It's not clear to me why this has to be done in a single thread?

(I also have the same question about `test_impossible_constraint`)

Reply:
Ah ok, because there is something checking that a warning is emitted.

Reply:
Indeed. https://py-free-threading.github.io/porting/#fixing-thread-unsafe-tests was helpful to understand the constructs that can't be used for parallel testing.